src/video/sgbpalettes.rs

```diff
@@ -95,190 +86,190 @@ const SGB_PALETTE_TABLE: &[(u8, u8, SGBPalette)] = &[
         make_colour!(0x00, 0x84, 0x00),
         make_colour!(0x00, 0x00, 0x00)
-    )
+    ]
 };
 
 const ZELDA: SGBPalette = SGBPalette {
-    bg: Matrix4::from_cols(
+    bg: [
         make_colour!(0xFF, 0xFF, 0xFF),
         make_colour!(0xFF, 0x84, 0x84),
         make_colour!(0x94, 0x3A, 0x3A),
         make_colour!(0x00, 0x00, 0x00)
-    ),
-    obj0: Matrix4::from_cols(
+    ],
+    obj0: [
         make_colour!(0xFF, 0xFF, 0xFF),
         make_colour!(0x00, 0xFF, 0x00),
         make_colour!(0x31, 0x84, 0x00),
         make_colour!(0x00, 0x4A, 0x00)
-    ),
-    obj1: Matrix4::from_cols(
+    ],
+    obj1: [
         make_colour!(0xFF, 0xFF, 0xFF),
         make_colour!(0x63, 0xA5, 0xFF),
         make_colour!(0x00, 0x00, 0xFF),
         make_colour!(0x00, 0x00, 0x00)
-    )
+    ]
 };
 
 const WARIO_LAND: SGBPalette = SGBPalette {
-    bg: Matrix4::from_cols(
+    bg: [
         make_colour!(0xFF, 0xFF, 0xFF),
         make_colour!(0xAD, 0xAD, 0x84),
         make_colour!(0x42, 0x73, 0x7B),
         make_colour!(0x00, 0x00, 0x00)
-    ),
-    obj0: Matrix4::from_cols(
+    ],
+    obj0: [
         make_colour!(0xFF, 0xFF, 0xFF),
         make_colour!(0xFF, 0x73, 0x00),
         make_colour!(0x94, 0x42, 0x00),
         make_colour!(0x00, 0x4A, 0x00)
-    ),
-    obj1: Matrix4::from_cols(
+    ],
+    obj1: [
         make_colour!(0xFF, 0xFF, 0xFF),
         make_colour!(0x5A, 0xBD, 0xFF),
         make_colour!(0xFF, 0x00, 0x00),
         make_colour!(0x00, 0x00, 0xFF)
-    )
+    ]
 };
 
 const DK_LAND: SGBPalette = SGBPalette {
-    bg: Matrix4::from_cols(
+    bg: [
         make_colour!(0xFF, 0xFF, 0x9C),
         make_colour!(0x94, 0xB5, 0xFF),
         make_colour!(0x63, 0x94, 0x73),
         make_colour!(0x00, 0x3A, 0x3A)
-    ),
-    obj0: Matrix4::from_cols(
+    ],
+    obj0: [
         make_colour!(0xFF, 0xC5, 0x42),
         make_colour!(0xFF, 0xD6, 0x00),
         make_colour!(0x94, 0x3A, 0x00),
         make_colour!(0x4A, 0x00, 0x00)
-    ),
-    obj1: Matrix4::from_cols(
+    ],
+    obj1: [
         make_colour!(0xFF, 0xFF, 0xFF),
         make_colour!(0xFF, 0x84, 0x84),
         make_colour!(0x94, 0x3A, 0x3A),
         make_colour!(0x00, 0x00, 0x00)
-    )
+    ]
 };
 
 const DK_LAND_2: SGBPalette = SGBPalette {
-    bg: Matrix4::from_cols(
+    bg: [
         make_colour!(0xFF, 0xFF, 0xFF),
         make_colour!(0x8C, 0x8C, 0xDE),
         make_colour!(0x52, 0x52, 0x8C),
         make_colour!(0x00, 0x00, 0x00)
-    ),
-    obj0: Matrix4::from_cols(
+    ],
+    obj0: [
         make_colour!(0xFF, 0xC5, 0x42),
         make_colour!(0xFF, 0xD6, 0x00),
         make_colour!(0x94, 0x3A, 0x00),
         make_colour!(0x4A, 0x00, 0x00)
-    ),
-    obj1: Matrix4::from_cols(
+    ],
+    obj1: [
         make_colour!(0xFF, 0xFF, 0xFF),
         make_colour!(0x5A, 0xBD, 0xFF),
         make_colour!(0xFF, 0x00, 0x00),
         make_colour!(0x00, 0x00, 0xFF)
-    )
+    ]
 };
 
 const MARIO_LAND: SGBPalette = SGBPalette {
```

Comment:
Why do you have hardcoded color palettes?

Reply:
So these are colour palettes for old game boy games that would appear by default on the GBC. They were actually hard coded into the GBC's internal ROM. This is why nintendo games like Pokemon Red & Blue have a very simple (non greyscale) palette. It can be overridden by passing in a different Palette to the rustboy when you construct it.